### PR TITLE
python3Packages.uxsim: 1.1.1 -> 1.3.0

### PR DESCRIPTION
**Pull request by vinnymeller**
## Description of changes

Updates [uxsim](https://github.com/toruseo/UXsim) to 1.3.0 & fixes python 3.12 build

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/uxsim/default.nix [hunk 1]
```diff
@@ -46,24 +45,20 @@ buildPythonPackage rec {
...
 
   # QT_PLUGIN_PATH is required to be set for the program to produce its images
   # our patch sets it to $NIX_QT_PLUGIN_PATH if QT_PLUGIN_PATH is not set
   # and here we replace this string with the actual path to qt plugins
   postInstall = ''
-    substituteInPlace $out/${python3.sitePackages}/uxsim/__init__.py \
+    substituteInPlace $out/${python.sitePackages}/uxsim/__init__.py \
       --replace-fail '$NIX_QT_PLUGIN_PATH' '${qt5.qtbase.bin}/${qt5.qtbase.qtPluginPrefix}'
-
-    mkdir -p $out/${python3.sitePackages}/uxsim/files
```
Comment (vinnymeller) [inline pkgs/development/python-modules/uxsim/default.nix:59]: font file was added upstream so don't need this anymore

File: pkgs/development/python-modules/uxsim/default.nix [hunk 2]
```diff
@@ -46,24 +45,20 @@ buildPythonPackage rec {
...
 
-  pythonImportsCheck = ["uxsim"];
-
+  pythonImportsCheck = [ "uxsim" ];
 
   # QT_PLUGIN_PATH is required to be set for the program to produce its images
   # our patch sets it to $NIX_QT_PLUGIN_PATH if QT_PLUGIN_PATH is not set
   # and here we replace this string with the actual path to qt plugins
   postInstall = ''
-    substituteInPlace $out/${python3.sitePackages}/uxsim/__init__.py \
```
Comment (vinnymeller) [inline pkgs/development/python-modules/uxsim/default.nix:56]: using `python3` here was causing the python3.12 build failure that I didn't notice

**Comment by vinnymeller** [comment]
Result of `nixpkgs-review pr 311372` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.uxsim</li>
    <li>python311Packages.uxsim.dist</li>
    <li>python312Packages.uxsim</li>
    <li>python312Packages.uxsim.dist</li>
  </ul>
</details>

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 311372` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.uxsim</li>
    <li>python311Packages.uxsim.dist</li>
    <li>python312Packages.uxsim</li>
    <li>python312Packages.uxsim.dist</li>
  </ul>
</details>

**Pull request merged**